arrays/900.js

```diff
@@ -0,0 +1,31 @@
+// One of these functions has side-effects.
+// When you call it, it does something other than just returning a value based only on its parameters.
+//
+// The other doesn't have side effects.
+// When you call it with the same arguments, it always does exactly the same thing,
+// and you can see everything it does in its return value.
+//
+// Which function has side-effects? Which doesn't?
+// Try to write tests for both functions.
```

Comment:
I think we need to include some extra info about target output / current output too so it will be easier to create tests for these functions. The mutation in the first implementation makes it less clear what this function is supposed to do.